### PR TITLE
[WIP]handler: add `/db-table?table_ids={...}`

### LINKED ISSUE #54281: When many tables under a DB, dashboard keyvis feature will cost a lot of resource
## Enhancement
img_v3_02c6_a75c806e-3a46-40ba-bdad-3cc752158b0g

### PR DESCRIPTION
**Pull request by lance6716**
### What problem does this PR solve?

Issue Number: close #54281

Problem Summary:

### What changed and how does it work?

```
curl http://{TiDBIP}:10080/db-table?table_ids={tableID,...}
```

### Check List

Tests 

- [ ] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: docs/tidb_http_api.md
Comment (lance6716) [inline docs/tidb_http_api.md:1]: wrong diff calculation 😂

File: pkg/server/http_status.go [hunk 1]
```diff
@@ -248,6 +248,7 @@ func (s *Server) startHTTPServer() {
 	router.Handle("/info", tikvhandler.NewServerInfoHandler(tikvHandlerTool)).Name("Info")
 	router.Handle("/info/all", tikvhandler.NewAllServerInfoHandler(tikvHandlerTool)).Name("InfoALL")
 	// HTTP path for get db and table info that is related to the tableID.
+	router.Handle("/db-table", tikvhandler.NewDBTableHandler(tikvHandlerTool)).Name("DB Table")
```
Comment (D3Hunter) [inline pkg/server/http_status.go:251]: prefer `/tables`, but we have used `/db-table`, ok to reuse the word

File: docs/tidb_http_api.md
Comment (lance6716) [reply docs/tidb_http_api.md:1]: 1. My IDE writes the correct order for the markdown item list
2. add the new API under "10. Get database information, table information and tidb info schema version by tableID."

File: pkg/server/handler/tests/http_handler_test.go [hunk 2]
```diff
@@ -923,6 +925,40 @@ func TestGetSchema(t *testing.T) {
...
+	require.NoError(t, resp.Body.Close())
+	require.Equal(t, 2, len(tis))
+	require.Equal(t, userTbl.Meta().ID, tis[int(userTbl.Meta().ID)].TableInfo.ID)
+	require.Equal(t, ddlTbl.Meta().ID, tis[int(ddlTbl.Meta().ID)].TableInfo.ID)
+
+	// check wrong format in table_ids
+	ids = ids + ",abc"
+	resp, err = ts.FetchStatus(fmt.Sprintf("/db-table?table_ids=%s", ids))
+	require.NoError(t, err)
+	require.NoError(t, resp.Body.Close())
```
Comment (D3Hunter) [inline pkg/server/handler/tests/http_handler_test.go:960]: check response code=400

File: pkg/server/handler/tikvhandler/tikv_handler.go [hunk 3]
```diff
@@ -1785,45 +1785,66 @@ type DBTableInfo struct {
...
+			if err != nil {
+				handler.WriteError(w, err)
+				return
+			}
+			tbl, err := getDBTableInfo(schema, int64(tid))
+			if err == nil {
+				data[tid] = tbl
+			}
+		}
+		handler.WriteData(w, data)
```
Comment (D3Hunter) [inline pkg/server/handler/tikvhandler/tikv_handler.go:1808]: maybe consider wrap data inside another common struct for further extend
```
{
   data: <real-data>
}
```

File: pkg/server/handler/tikvhandler/tikv_handler.go [hunk 4]
```diff
@@ -1785,45 +1785,66 @@ type DBTableInfo struct {
...
+		data := make(map[int]*DBTableInfo, len(tableIDs))
+		for _, tableID := range tableIDs {
+			tid, err := strconv.Atoi(tableID)
+			if err != nil {
+				handler.WriteError(w, err)
+				return
+			}
+			tbl, err := getDBTableInfo(schema, int64(tid))
+			if err == nil {
+				data[tid] = tbl
```
Comment (D3Hunter) [inline pkg/server/handler/tikvhandler/tikv_handler.go:1805]: do dashboard need to know it's partially result?

**Review by D3Hunter — COMMENTED** [review]
rest lgtm

File: pkg/server/handler/tikvhandler/tikv_handler.go (hunk 4 above)
Comment (lance6716) [reply pkg/server/handler/tikvhandler/tikv_handler.go:1805]: Current way to process the result is verified by dashboard dev.

File: pkg/server/handler/tikvhandler/tikv_handler.go (hunk 3 above)
Comment (lance6716) [reply pkg/server/handler/tikvhandler/tikv_handler.go:1808]: @mornyx Do you think it's better we add another level JSON object?

Comment (mornyx) [reply pkg/server/handler/tikvhandler/tikv_handler.go:1808]: `/schema` does not follow this design. Maybe we should keep them consistent?

File: pkg/server/handler/tikvhandler/tikv_handler.go (hunk 4 above)
Comment (mornyx) [reply pkg/server/handler/tikvhandler/tikv_handler.go:1805]: For the dashboard scenario alone this is as designed.

**Review by D3Hunter — APPROVED** [review]

**Comment by tiancaiamao** [comment]
If it's ready for review, you can remove the [WIP] label @lance6716

**Comment by lance6716** [comment]
Due to problems on keyvis side, I will open another PR to fix the issue. The API in this PR can't be used smoothly.

**Comment by lance6716** [comment]
replaced by https://github.com/pingcap/tidb/pull/54608 (still WIP)

**Pull request closed without merging**